test/cpp/CMakeLists.txt

```diff
@@ -1,5 +1,9 @@
 cmake_minimum_required(VERSION 3.0)
 
+if( NOT CXX )
+  set(CMAKE_CXX_COMPILER clang++)
```

Comment:
why is this needed?

Reply:
This is for the upstream test -- they have different compiler flags and settings. e.g., we set `CXX=clang-8` which is not common :)